lib/spack/spack/util/web.py

```diff
@@ -85,6 +86,27 @@ def handle_starttag(self, tag, attrs):
                     self.links.append(val)
 
 
+def read_etag(url):
```

Comment:
See spack.util.web.parse_etag. Can you use that to guard against garbage etag responses?

Reply:
yeah, I saw the double quotes getting pulled in. I didn't see the `parse_etag` API, I will use that, thanks!